doc/introduction/new_opmath.rst

```diff
@@ -0,0 +1,424 @@
+To help identify a fix, select the option below that describes your situation.
+
+.. details::
+    :title: My old PennyLane script does not run anymore
+    :href: old-script-broken
+
+    A quick-and-dirty fix for this issue is to deactivate new opmath at the beginning of the script via ``qml.operation.disable_new_opmath()``.
+    We recommend to do the following checks instead
+
+    * Check explicit use of the legacy :class:`~Tensor` class. If you find it in your script it can just be changed from ``Tensor(*terms)`` to ``qml.prod(*terms)`` with the same call signature.
+
+    * Check explicit use of ``op.obs`` attribute, where ``op`` is some operator. This is how the terms of a tensor product is accessed in :class:`~Tensor` instances. Use ``op.operands`` instead.
+
+      .. code-block:: python
+
+          # new opmath enabled (default)
+          op = X(0) @ X(1)
+          assert op.operands == (X(0), X(1))
+
+          with qml.operation.disable_new_opmath_cm():
+              # context manager that disables new opmath temporarilly
+              op = X(0) @ X(1)
+              assert op.obs == [X(0), X(1)]
+    
+    * Check explicit use of ``qml.ops.Hamiltonian``. In that case, simply change to ``qml.Hamiltonian``.
+
+      >>> op = qml.ops.Hamiltonian([0.5, 0.5], [X(0) @ X(1), X(1) @ X(2)])
+      ValueError: Could not create circuits. Some or all observables are not valid.
+      >>> op = qml.Hamiltonian([0.5, 0.5], [X(0) @ X(1), X(1) @ X(2)])
+      >>> isinstance(op, qml.ops.LinearCombination)
+      True
+    
+    * Check if you are explicitly enabling and disabling new opmath somewhere in your script. Mixing both systems is not supported.
+
+    If for some unexpected reason your script still breaks, please post on the PennyLane `discussion forum <https://discuss.pennylane.ai>`_ or open a
+    `bug report <https://github.com/PennyLaneAI/pennylane/issues/new?labels=bug+%3Abug%3A&template=bug_report.yml&title=[BUG]>`_
+    on the PennyLane GitHub page.
+
+.. details::
+    :title: Sharp bits about the qml.Hamiltonian dispatch
+    :href: sharp-bits-hamiltonian
+
+    One of the reasons that :class:`~ops.LinearCombination` exists is that the old Hamiltonian class is not compatible with new opmath tensor products.
+    We can try to instantiate a old ``qml.ops.Hamiltonian`` class with a ``X(0) @ X(1)`` tensor product, which returns a :class:`~Prod` instance with opmath enabled.
+
+    >>> qml.operation.active_new_opmath() # confirm opmath is active (by default)
+    True
+    >>> qml.ops.Hamiltonian([0.5], [X(0) @ X(1)])
+    PennyLaneDeprecationWarning: Using 'qml.ops.Hamiltonian' with new operator arithmetic is deprecated. Instead, use 'qml.Hamiltonian', or use 'qml.operation.disable_new_opmath()' to continue to access the legacy functionality. See https://docs.pennylane.ai/en/stable/development/deprecations.html for more details.
+    ValueError: Could not create circuits. Some or all observables are not valid.

```

Comment:
This feels like not the sort of error I was expecting :thinking: Where is the circuit?

Reply:
I think this is just an unfortunate phrasing in the `Hamiltonian` class in https://github.com/PennyLaneAI/pennylane/blob/master/pennylane/ops/qubit/hamiltonian.py#L208 Would be an easy fix on the PennyLane side but I am afraid this would warrant also changes to multiple dependency unit tests